src/script/hydra-eval-jobset

```diff
@@ -809,41 +809,6 @@ sub checkJobsetWrapped {
                 $ev->jobsetevalmembers->create({ build => $id, isnew => $x->{new} });
             }
 
-            # Create AggregateConstituents mappings.  Since there can
-            # be jobs that alias each other, if there are multiple
-            # builds for the same derivation, pick the one with the
-            # shortest name.
-            my %drvPathToId;
-            foreach my $id (keys %buildMap) {
-                my $x = $buildMap{$id};
-                my $y = $drvPathToId{$x->{drvPath}};
-                if (defined $y) {
-                    next if length $x->{jobName} > length $y->{jobName};
-                    next if length $x->{jobName} == length $y->{jobName} && $x->{jobName} ge $y->{jobName};
-                }
-                $drvPathToId{$x->{drvPath}} = $x;
-            }
-
-            # XXX: dead code with nix-eval-jobs. To be removed.
-            foreach my $job (values @jobs) {
-                next unless $job->{constituents};
-
-                if (defined $job->{error}) {
-                    die "aggregate job ‘$job->{jobName}’ failed with the error: $job->{error}\n";
-                }
-
-                my $x = $drvPathToId{$job->{drvPath}} or
-                    die "aggregate job ‘$job->{jobName}’ has no corresponding build record.\n";
-                foreach my $drvPath (@{$job->{constituents}}) {
-                    my $constituent = $drvPathToId{$drvPath};
-                    if (defined $constituent) {
-                        $db->resultset('AggregateConstituents')->update_or_create({aggregate => $x->{id}, constituent => $constituent->{id}});
```

Comment:
@Ericson2314 are we depending on this in hydra.nixos.org?

Reply:
According to the commit message we are not.

But good mentoring this.

Reply:
I find the statement that we weren't using this interesting, nixpkgs uses it in both code and documentation:
```
doc/preface.chapter.md:[this table](https://hydra.nixos.org/job/nixpkgs/trunk/unstable#tabs-constituents)
lib/customisation.nix:          constituents = map hydraJob (flatten drv.constituents);
nixos/doc/manual/release-notes/rl-2305.section.md:- Deprecated `xlibsWrapper` transitional package has been removed in favour of direct use of its constituents: `xorg.libX11`, `freetype` and others.
nixos/release-combined.nix:      constituents = pkgs.lib.concatLists [
nixos/release-small.nix:    constituents = lib.flatten [
pkgs/build-support/release/default.nix:    { name, constituents, meta ? { } }:
pkgs/build-support/release/default.nix:      { inherit constituents meta;
pkgs/build-support/release/default.nix:        echo $constituents > $out/nix-support/hydra-aggregate-constituents
pkgs/build-support/release/default.nix:        for i in $constituents; do
pkgs/build-support/release/default.nix:     - constituents: a list of derivations on which the channel success depends.
pkgs/build-support/release/default.nix:         constituents = [ foo bar baz ];
pkgs/build-support/release/default.nix:    { name, src, constituents ? [], meta ? {}, isNixOS ? true, ... }@args:
pkgs/build-support/release/default.nix:        echo $constituents > "$out/nix-support/hydra-aggregate-constituents"
pkgs/build-support/release/default.nix:        for i in $constituents; do
pkgs/build-support/rust/build-rust-crate/test/default.nix:    constituents = builtins.attrValues tests;
pkgs/top-level/release-haskell.nix:  # to `releaseTools.aggregate` as constituents.
pkgs/top-level/release-haskell.nix:        constituents =
pkgs/top-level/release-haskell.nix:        constituents = accumulateDerivations
pkgs/top-level/release-haskell.nix:        constituents = accumulateDerivations [
pkgs/top-level/release-haskell.nix:        constituents = accumulateDerivations [
pkgs/top-level/release.nix:          constituents = [
pkgs/top-level/release.nix:          constituents =
pkgs/top-level/release-python.nix:      constituents = [
README.md:* [Tests for unstable/master](https://hydra.nixos.org/job/nixos/trunk-combined/tested#tabs-constituents)
README.md:* [Tests for the NixOS 24.05 release](https://hydra.nixos.org/job/nixos/release-24.05/tested#tabs-constituents)
```

Reply:
That sounds like a blocking issue to me

Reply:
https://github.com/nix-community/nix-eval-jobs/commit/a4441fe1c8949665a6128ef9224def7a04dbd211 This is the commit we'd need to revert of `nix-eval-jobs`, I guess. Hmm...

Reply:
Hmm, so there are multiple types of constituent jobs. I am now less sure that we are still using this type.

Reply:
Constituents are back now, thanks @Ma27!